src/session.ts

```diff
@@ -268,5 +298,26 @@ export class Session {
 
         return result
     }
+    validateBeforeSignSignatures(
+        context: TransactContext,
+        request: SigningRequest,
+        result: TransactResult
+    ): void {
+        const requestTransaction = request.getRawTransaction()
+        const requestDigest = requestTransaction.signingDigest(this.chain.id)
+        const publicKeys = Serializer.objectify(result.keys)
+        result.signatures.forEach((signature) => {
+            const recoveredKey = signature.recoverDigest(requestDigest)
+            const verified = signature.verifyDigest(requestDigest, recoveredKey)
+            if (!verified || !publicKeys.includes(String(recoveredKey))) {
+                throw new Error(
+                    `A signature (${signature}) provided by a beforeSign hook using ` +
+                        `a key (${recoveredKey}) has been invalidated, likely due to the transaction ` +
+                        `being modified by another hook after the signature was created. To disable ` +
+                        `this error, set validatePluginSignatures equal to false.`
+                )
+            }
+        })
+    }
```

Comment:
This method takes the current request and all public keys recovered from the signatures provided by the `beforeSign` hooks. It recovers the digest from the transaction, and then for each signature, ensures that the signature is valid for the digest and the key recovered from the current digest/signature combo matches what was originally used.